Core/src/org/sleuthkit/autopsy/discovery/ui/MiniTimelineArtifactListPanel.java

```diff
@@ -0,0 +1,265 @@
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    void addSelectionListener(ListSelectionListener listener) {
+        jTable1.getSelectionModel().addListSelectionListener(listener);
+    }
+
+    /**
+     * Remove a listener from the table of dates.
+     *
+     * @param listener The listener to remove from the table of dates.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    void removeListSelectionListener(ListSelectionListener listener) {
+        jTable1.getSelectionModel().removeListSelectionListener(listener);
+    }
+
+    /**
+     * Whether the list of dates is empty.
+     *
+     * @return True if the list of dates is empty, false if there are dates.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    boolean isEmpty() {
+        return tableModel.getRowCount() <= 0;
+    }
+
+    /**
+     * Select the first available date in the list if it is not empty to
+     * populate the list to the right.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    void selectFirst() {
+        if (!isEmpty()) {
+            jTable1.setRowSelectionInterval(0, 0);
+        } else {
+            jTable1.clearSelection();
+        }
+    }
+
+    /**
+     * The artifact which is currently selected, null if no artifact is
+     * selected.
+     *
+     * @return The currently selected BlackboardArtifact or null if none is
+     *         selected.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    BlackboardArtifact getSelectedArtifact() {
+        int selectedIndex = jTable1.getSelectionModel().getLeadSelectionIndex();
+        if (selectedIndex < jTable1.getSelectionModel().getMinSelectionIndex() || jTable1.getSelectionModel().getMaxSelectionIndex() < 0 || selectedIndex > jTable1.getSelectionModel().getMaxSelectionIndex()) {
+            return null;
+        }
+        return tableModel.getArtifactByRow(jTable1.convertRowIndexToModel(selectedIndex));
+    }
+
+    /**
+     * Add the specified list of dates to the list of dates which should be
+     * displayed.
+     *
+     * @param dateCountList The list of dates to display.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    void addArtifacts(List<BlackboardArtifact> dateCountList) {
+        tableModel.setContents(dateCountList);
+        jTable1.validate();
+        jTable1.repaint();
+        tableModel.fireTableDataChanged();
+    }
+
+    /**
+     * Remove all artifacts from the list of artifacts displayed.
+     */
+    @ThreadConfined(type = ThreadConfined.ThreadType.AWT)
+    void clearDates() {
+        tableModel.setContents(new ArrayList<>());
+        tableModel.fireTableDataChanged();
+    }
+
+    /**
+     * Initialize the UI components.
+     */
+    private void initComponents() {
+        //This class is a refactored copy of ArtifactsListPanel so lacks the form however the init method still constructs the proper UI elements.
```

Comment:
It was a copy refactor so there is not a form file. I did put this comment in the initComponents file to reflect the nature of this file.

Reply:
Yeah, I missed that.